x-pack/legacy/plugins/apm/server/routes/settings.ts

```diff
@@ -4,220 +4,189 @@
+  method: 'DELETE',
+  path: '/api/apm/settings/agent-configuration/{configurationId}',
+  params: {
+    path: t.type({
+      configurationId: t.string
+    }),
+    query: minimalRt
+  },
+  handler: async (req, { path }) => {
+    const setup = await setupRequest(req);
+    const { configurationId } = path;
+    return await deleteConfiguration({
+      configurationId,
+      setup
+    });
+  }
+}));
+
+// get list of services
+export const listAgentConfigurationServicesRoute = createRoute(() => ({
+  method: 'GET',
+  path: '/api/apm/settings/agent-configuration/services',
+  params: {
+    query: minimalRt
+  },
+  handler: async req => {
+    const setup = await setupRequest(req);
+    return await getServiceNames({
+      setup
+    });
+  }
+}));
+
+// get environments for service
+export const listAgentConfigurationEnvironmentsRoute = createRoute(() => ({
+  path:
+    '/api/apm/settings/agent-configuration/services/{serviceName}/environments',
+  params: {
+    path: t.type({
+      serviceName: t.string
+    }),
+    query: minimalRt
+  },
+  handler: async (req, { path }) => {
+    const setup = await setupRequest(req);
+    const { serviceName } = path;
+    return await getEnvironments({
+      serviceName,
+      setup
+    });
+  }
+}));
+
+// create configuration
+const transactionSampleRateRt = new t.Type<number, number, unknown>(
+  'TransactionSampleRate',
+  (u): u is number => typeof u === 'number',
+  (u, c) => {
+    return Number(u) >= 0 && Number(u) <= 1
+      ? t.success(Number(u))
+      : t.failure(u, c);
+  },
+  a => Number(a.toPrecision(3))
+);
+
+const agentPayloadRt = t.type({
+  settings: t.type({
+    transaction_sample_rate: transactionSampleRateRt
+  }),
+  service: t.intersection([
+    t.type({ name: t.string }),
+    t.partial({ environment: t.string })
+  ])
+});
+
+export const createAgentConfigurationRoute = createRoute(() => ({
+  method: 'POST',
+  path: '/api/apm/settings/agent-configuration/new',
+  params: {
+    body: agentPayloadRt,
+    query: minimalRt
+  },
+  handler: async (req, { body }) => {
+    const setup = await setupRequest(req);
+    return await createConfiguration({
+      configuration: body,
+      setup
+    });
+  }
+}));
+
+export const updateAgentConfigurationRoute = createRoute(() => ({
+  method: 'PUT',
+  path: `/api/apm/settings/agent-configuration/{configurationId}`,
+  params: {
+    path: t.type({
+      configurationId: t.string
     }),
-    service: Joi.object({
-      name: Joi.string().required(),
-      environment: Joi.string()
+    body: agentPayloadRt,
+    query: minimalRt
+  },
+  handler: async (req, { path, body }) => {
+    const setup = await setupRequest(req);
+    const { configurationId } = path;
+    return await updateConfiguration({
+      configurationId,
+      configuration: body,
+      setup
+    });
+  }
+}));
+
+// Lookup single configuration
+const searchRoute = createRoute(core => ({
+  method: 'POST' as const,
```

Comment:
Do you need `const` here? I don't see it anywhere else.

Reply:
No, don't think I need it anymore, possibly a left over from an earlier implementation. Will remove